### PR TITLE
Feature/wildcard support

### PR DESCRIPTION
**Pull request by irfanshar**

### CODE REVIEW

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]
This is great! do you have any tests defined?  Say callng rules_mk.from_str and testing the results RulesMk object?

File: src/makei/rules_mk.py [hunk 1]
```diff
@@ -232,14 +233,30 @@ def from_str(cls, rules_mk_str: str, containing_dir: Path, src_dir: Path, includ
                 subdir = line.strip().split('=')[1].split()
                 continue
             elif ':' in line:
+                # rules_mk variable definition
```
Comment (edmundreinhardt) [inline src/makei/rules_mk.py:240]: let's put and example here to illustrate i.e.
# var := value

File: src/makei/rules_mk.py [hunk 2]
```diff
@@ -266,9 +267,17 @@ def from_str(cls, rules_mk_str: str, containing_dir: Path, src_dir: Path, includ
 
                     variables_to_process[key].append(variable)
                 else:
-                    # recipe
-                    recipe_env = True
-                    recipe_str = line + '\n'
+                    # Wildcard %.TARGET: %.SOURCE DEP1 DEP2
```
Comment (edmundreinhardt) [inline src/makei/rules_mk.py:270]: Test if we handle wildcard recipes that span lines i.e.
```
%.MODULE: %.RPGLE DEP1.FILE \
                      DEP2.FILE

File: src/makei/rules_mk.py [hunk 3]
```diff
@@ -278,12 +287,24 @@ def from_str(cls, rules_mk_str: str, containing_dir: Path, src_dir: Path, includ
             targets.append(recipe_str.split(":")[0])
             rules.append(MKRule.from_str(recipe_str, containing_dir, include_dirs))
 
+        # Create all the rules for the wildcard rule declaration
+        for target_ext, object_ext, dependencies in wildcard_targets:
```
Comment (edmundreinhardt) [inline src/makei/rules_mk.py:291]: second in list should be named `source_ext` not `object_ext`

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]
We need to add tests

**Comment by edmundreinhardt** [comment]
Tested the following to prove that we can't assume that ':=' will be space delimited
```
CURRENT:=V7R5

all: 
	echo $(CURRENT)
```
gmake all produces `V7R5` therefore we need to support this

**Comment by SanjulaGanepola** [comment]
@edmundreinhardt Can you review this PR and suggest any more edge cases to test. I resolved the test case issues we found earlier. All tests are now passing and bob-recursive-example build is still good. If all looks good, we can merge and release.

image
image

File: src/makei/rules_mk.py [hunk 4]
```diff
@@ -232,20 +232,21 @@ def from_str(cls, rules_mk_str: str, containing_dir: Path, src_dir: Path, includ
                 continue
 
             # pylint: disable=no-else-continue
+            pattern = re.compile(r'^[^:]*(:=|=).*')
```
Comment (edmundreinhardt) [inline src/makei/rules_mk.py:251]: add a comment what regex

File: tests/data/rules_mks/wildcard/wildcard.rules.mk [hunk 5]
```diff
@@ -1,6 +1,6 @@
 TGTVER := *PRV
 # need to support when no blanks delimiting the equals
-CURRENT := V7R5
+CURRENT:=V7R5
```
Comment (edmundreinhardt) [inline tests/data/rules_mks/wildcard/wildcard.rules.mk:3]: also test 'foo=val' for regular =

**Comment by SanjulaGanepola** [comment]
@edmundreinhardt I addressed the comments. I believe some of the comments you were going to leave when we were reviewing did not get posted here. I think I caught everything however. Let me know if anything was missed.

**Review by edmundreinhardt — APPROVED** [review]
